Client/src/Common/Submitter/SessionService.cs

```diff
@@ -46,68 +48,162 @@
-  {
-    channelPool_     = channelPool;
-    Logger           = loggerFactory?.CreateLogger<T>();
-    chunkSubmitSize_ = chunkSubmitSize;
-  }
+  public readonly ChannelPool ChannelPool;
+
+  private readonly int chunkSubmitSize_;
 
   /// <summary>
-  ///   Set or Get TaskOptions with inside MaxDuration, Priority, AppName, VersionName and AppNamespace
+  ///   Logger
   /// </summary>
-  public TaskOptions TaskOptions { get; set; }
+  protected readonly ILogger Logger;
 
   /// <summary>
-  ///   Get SessionId object stored during the call of SubmitTask, SubmitSubTask,
-  ///   SubmitSubTaskWithDependencies or WaitForCompletion, WaitForSubTaskCompletion or GetResults
+  ///   Default task options of the session
   /// </summary>
-  public Session SessionId { get; protected set; }
-
-
-#pragma warning restore CS1591
+  public readonly TaskOptions TaskOptions;
 
   /// <summary>
-  ///   The channel pool to use for creating clients
+  ///   Creates a session service
   /// </summary>
-  protected ChannelPool channelPool_;
+  /// <param name="channelPool">Grpc channel pool</param>
+  /// <param name="sessionTaskOptions">Default task options for the session</param>
+  /// <param name="loggerFactory"></param>
+  /// <param name="session"></param>
+  /// <param name="chunkSubmitSize"></param>
+  public SessionService(ChannelPool                channelPool,
+                        TaskOptions                sessionTaskOptions,
+                        [CanBeNull] ILoggerFactory loggerFactory   = null,
+                        [CanBeNull] Session        session         = null,
+                        int                        chunkSubmitSize = 500)
+  {
+    Logger           = loggerFactory?.CreateLogger<SessionService>();
+    ChannelPool      = channelPool;
+    chunkSubmitSize_ = chunkSubmitSize;
+    TaskOptions      = sessionTaskOptions;
+
+    SessionId = session is null
+                  ? CreateSession(TaskOptions)
+                  : OpenSession(session);
 
+    Logger?.LogDebug("Session {Status} {SessionId}",
+                     session is not null
+                       ? "Opened"
+                       : "Created",
+                     SessionId);
+  }
 
   /// <summary>
-  ///   The number of chunk to split the payloadsWithDependencies
+  ///   Current session id
   /// </summary>
-  private int chunkSubmitSize_;
+  public Session SessionId { get; private set; }
 
   /// <summary>
-  ///   The logger to call the generate log in Seq
+  ///   Get the default task options for a given engine type
   /// </summary>
+  /// <param name="engineType">Engine type</param>
+  /// <returns>Default task options for the given engine type</returns>
+  [PublicAPI]
+  public static TaskOptions GetDefaultTaskOptions(EngineType engineType)
+    => new()
+       {
+         MaxDuration = new Duration
+                       {
+                         Seconds = 300,
+                       },
+         MaxRetries = 3,
+         Priority   = 1,
```

Comment:
I think a Priority of 2 would be better by default (2 is neither the minimum, nor the maximum).